File: 0788-Rotated Digits/0788-Rotated Digits.cpp
class Solution {
public:
    int rotatedDigits(int N) {
        int count = 0;
        for (int i = 1; i <= N; ++i) {
            if (isValid(i)) {
                ++count;
            }
        }
        return count;
    }
    
private:
    bool isValid(int num) {
        int n = num, base = 1;
        int result = 0;
        while (n) {
            int temp = n % 10;
            if (temp == 2 || temp == 5) {
                temp = 7 - temp;
            }
            else if (temp == 6 || temp == 9) {
                temp = 15 - temp;
            }
            else if (temp != 0 && temp != 1 && temp != 8) {
                return false;
            }
            result += base * temp;
            base *= 10;
            n /= 10;
        }
        return result != num;
    }
};

File: 0701-0800/0788-Rotated Digits/0788-Rotated Digits.py
class Solution:
    def rotatedDigits(self, N: int) -> int:
        count = 0
        for i in range(1, N + 1):
            s = str(i)
            count += all(c not in '347' for c in s) and any(c in '2569' for c in s)
        return count
